inst/@sym/solve.m

```diff
@@ -116,35 +137,22 @@
             '    d = d[0]'
             'return d,' };
 
-    out = python_cmd (cmd, varargin{:});
+    out = python_cmd ([strrep(base, '{extra}', ', dict=True'); cmd], varargin{:});
 
     varargout = {out};
 
   else  % multiple outputs
-    cmd = { 'eqs = list(); symbols = list()'
-            'for arg in _ins:'
-            '    if arg.is_Relational:'
-            '        eqs.append(arg)'
-            '    else:'
-            '        symbols.append(arg)'
-            '#'
-            'if len(symbols) > 0:'
-            '    (vars, solns) = sp.solve(eqs, symbols, set=True)'
-            'else:'
-            '    (vars, solns) = sp.solve(eqs, set=True)'
-            '#'
-            'd = []'
+    cmd = { '(vars, solns) = d'
+            'q = []'
             'for (i, var) in enumerate(vars):'
-            '    d.append(sp.Matrix([t[i] for t in solns]))'
-            'return d,' };
+            '    q.append(sp.Matrix([t[i] for t in solns]))'
+            'return q,' };
 
-    out = python_cmd (cmd, varargin{:});
+    out = python_cmd ([strrep(base, '{extra}', ', set=True'); cmd], varargin{:});
 
-    varargout = out;
+    assert( isequal( nargout, length(out)), ['This system have ' num2str(length(out)) ' solution, favor set correct numbers of outputs.']);
 
-    if (length(out) ~= nargout)
-      warning('solve: number of outputs did not match solution vars');
-    end
```

Comment:
mm, i think its a little of the last two things, basically the first thing its octave send an error, but you know, that don't have to much info, so with the assert we can say to user the problem.
in other words, when the requested values don't math with out we have the warning, but then octave send the error.., at least to me don't have to many sense, we can just send the error with the info.